lib/questions.rb

```diff
@@ -63,33 +84,47 @@ def separate_array_into_even_and_odd_numbers(array)
 # e.g. 'bob'. So in the array ['bob', 'radar', 'eat'], there
 # are 2 palindromes (bob and radar), so the method should return 2
 def number_of_elements_that_are_palindromes(array)
+  new = array.push(array.map{|array| array.reverse}).flatten
+  new.length - new.uniq.length
 end
 
 # return the shortest word in an array
 def shortest_word_in_array(array)
+  shortest = array.inject do |memo, word|
+  memo.length < word.length ? memo : word
+  end
 end
 
 # return the shortest word in an array
 def longest_word_in_array(array)
+  shortest = array.inject do |memo, word|
+  memo.length > word.length ? memo : word
+  end
 end
 
 # add up all the numbers in an array, so [1, 3, 5, 6]
 # returns 15
 def total_of_array(array)
+  array.inject{|array, sum| sum + array }
 end
 
 # turn an array into itself repeated twice. So [1, 2, 3]
 # becomes [1, 2, 3, 1, 2, 3]
 def double_array(array)
+  new = []
+  (array.map{|array| new << array})[0..1].flatten
```

Comment:
Shadowing outer local variable - `array`.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.